servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/versionstore/ConfigurableVersionStoreFactory.java

```diff
@@ -96,22 +87,6 @@ public VersionStore getVersionStore() {
       if (storeConfig.isEventsEnabled() && resultConsumer.isResolvable()) {
         versionStore = new EventsVersionStore(versionStore, resultConsumer.get());
       }
-      if (storeConfig.isTracingEnabled()) {
-        if (opentelemetryTracer.isUnsatisfied()) {
-          LOGGER.warn(
-              "OpenTelemetry is enabled, but not available, forgot to add quarkus-opentelemetry?");
-        } else {
-          Tracer t = opentelemetryTracer.get();
-          versionStore = new TracingVersionStore(t, versionStore);
-        }
-      }
-      if (storeConfig.isMetricsEnabled()) {
-        if (meterRegistry.isUnsatisfied()) {
-          LOGGER.warn("Metrics are enabled, but not available, forgot to add quarkus-micrometer?");
-        } else {
-          versionStore = new MetricsVersionStore(versionStore, meterRegistry.get(), Clock.SYSTEM);
-        }
-      }
       return versionStore;
```

Comment:
The usage of the `@NotObserved` qualifier is certainly clever, but wouldn't it be much simpler to just:

```java
      return new ObservingVersionStore(versionStore);
```

Since the `ObservingVersionStore` bean will always be created anyways, and is the only one injectable on endpoints expecting a `@Default` bean?

Reply:
IIRC the annotated metrics/tracing would not work :(

Reply:
Hmm OK the annotations must be on a managed bean indeed.